File: java/src/main/java/it/giannibombelli/wsc2026/giftcard/application/commands/CreditGiftCard.java
package it.giannibombelli.wsc2026.giftcard.application.commands;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.application.Command;
import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import it.giannibombelli.wsc2026.giftcard.domain.giftcard.GiftCardId;

public record CreditGiftCard(GiftCardId aggregateId, Money amount) implements Command<GiftCardId> {
    public CreditGiftCard {
        Require.requireArgument(aggregateId, "giftCardId");
        Require.requireArgument(amount, "amount");
    }
}

File: typescript/src/giftcard/application/commands/creditGiftCard.ts
// Command per accreditare una gift card.

import type { Command } from '@/common/application/command.js';
import type { Money } from '@/common/domain/primitive/money.js';
import type { GiftCardId } from '../../domain/giftcard/giftCardId.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type CreditGiftCard = Command<'CreditGiftCard', GiftCardId> & {
  readonly amount: Money;
};

export function creditGiftCard(aggregateId: GiftCardId, amount: Money): CreditGiftCard {
  requireArgument(aggregateId, 'giftCardId');
  requireArgument(amount, 'amount');
  return { kind: 'CreditGiftCard', aggregateId, amount };
}
